code/server/test/api_endpoint_tests.js

```diff
@@ -1,26 +1,29 @@
 var app = require('../main.js');
 var request = require('supertest');
 var assert = require('assertthat'); // View README for documentation https://github.com/thenativeweb/assertthat
-
+var api = require('../routes/api.js');
+var stubUser = require('../stub_models/StubUser.js');
+var stubLikes = require('../stub_models/StubUserMatches.js');
 //////////////////////////////////////////
 // ALL API TESTS SHOULD GO IN THIS FILE // 
 //////////////////////////////////////////
 
 describe('api', function() {
+          api.injectUser(stubUser);
+          api.injectLikes(stubLikes);
```

Comment:
Can you explain to me how this works? Importing the api file in the test file doesn't make sense to me, because it's imported in main.js (in the app variable) which mocha uses to make the get/post requests. It just seems like it's a duplicate import and I can't understand how stubbing the api variable that you created is carrying over to the api imported by main.js.